### PR TITLE
#1494: Save image content hash to the media asset when saving image from the Adobe Stock

### LINKED ISSUE #1494: Save image content hash to the media asset when saving image from the Adobe Stock
Story: https://github.com/magento/adobe-stock-integration/issues/1484

### Preconditions (*)
1. Install Magento 2.4-develop with ASI 2.0-develop
2. Enable and Configure Adobe Stock Integration in the Stores -> Configuration -> System -> Adobe Stock Integration

### Steps to reproduce (*)
1. Login to Admin panel
2. Open the media gallery. I.e. from the Catalog -> Category -> Content fieldset -> Select image from gallery
3. Click "Search Adobe Stock" button
4. Click on any image to open the image preview panel
5. Click "Save Preview" button
6. Confirm the image name in the popup

### Expected result (*)
The images information should be saved to the `media_gallery_asset` table. The `hash` filed should be populated with the `sha1` hash of the image content

### Additional info

The same use case should work for Licensing or Saving Licensed image

### PR DESCRIPTION
**Pull request by joweecaquicla**
### Description (*)

This PR will implement the saving of image content hash to the media asset.
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/adobe-stock-integration/issues/1494: Save image content hash to the media asset when saving image from the Adobe Stock 
2. ...

### Manual testing scenarios (*)

1. Login to Admin panel
2. Open the media gallery. I.e. from the Catalog -> Category -> Content fieldset -> Select image from gallery
3. Click "Search Adobe Stock" button
4. Click on any image to open the image preview panel
5. Click "Save Preview" button
6. Confirm the image name in the popup

### Expected result (*)
The images information should be saved to the `media_gallery_asset` table. The hash filed should be populated with the `sha1` hash of the image content

### CODE REVIEW

**Comment by joweecaquicla** [comment]
@magento run all tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @joweecaquicla ! Looks good! Can you please fix the failed unit test

**Comment by joweecaquicla** [comment]
@magento run all tests

File: AdobeStockImage/Model/SaveMediaGalleryAsset.php [hunk 1]
```diff
@@ -86,4 +89,19 @@ private function calculateFileSize(string $destinationPath): int
...
+     * @return string
+     * @throws FileSystemException
+     */
+    private function hashImageContent(string $destinationPath): string
+    {
+        $mediaDirectory = $this->fileSystem->getDirectoryRead(DirectoryList::MEDIA);
+        $imageContent = $mediaDirectory->readFile($mediaDirectory->getAbsolutePath($destinationPath));
+        $hashedImageContent = sha1($imageContent);
+        return $hashedImageContent;
+    }
```
Comment (coderimus) [inline AdobeStockImage/Model/SaveMediaGalleryAsset.php]:
```suggestion
    private function hashImageContent(string $destinationPath): string
    {
        return sha1_file($destinationPath);
    }
```
cc @sivaschenko

File: AdobeStockImage/Model/SaveMediaGalleryAsset.php [hunk 2]
```diff
@@ -61,11 +62,13 @@ public function execute(Document $document, string $destinationPath): void
     {
         try {
             $fileSize = $this->calculateFileSize($destinationPath);
+            $hashedImageContent = $this->hashImageContent($destinationPath);
             $additionalData = [
                 'id' => null,
                 'path' => $destinationPath,
                 'source' => 'Adobe Stock',
                 'size' => $fileSize,
+                'hash' => $hashedImageContent
```
Comment (coderimus) [inline AdobeStockImage/Model/SaveMediaGalleryAsset.php]:
```suggestion
                'hash' => $this->hashImageContent($destinationPath)
```

File: AdobeStockImage/Model/SaveMediaGalleryAsset.php [hunk 3]
```diff
@@ -61,11 +62,13 @@ public function execute(Document $document, string $destinationPath): void
     {
         try {
             $fileSize = $this->calculateFileSize($destinationPath);
+            $hashedImageContent = $this->hashImageContent($destinationPath);
```
Comment (coderimus) [inline AdobeStockImage/Model/SaveMediaGalleryAsset.php]: We can use the `hashImageContent` directly when assigning value to the array key. Please, remove this line.

**Review by coderimus — CHANGES_REQUESTED** [review]
Hi @joweecaquicla ! Thank you for the PR provided 👍 Please, check my change requests below.

File: AdobeStockImage/Model/SaveMediaGalleryAsset.php (hunk 1 above)
Comment (coderimus) [reply AdobeStockImage/Model/SaveMediaGalleryAsset.php]: Read the comment https://github.com/magento/adobe-stock-integration/pull/1510#discussion_r444850084 and see the point with the `sha1_file` @joweecaquicla please, ignore my change request.

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by joweecaquicla** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**